### PR TITLE
[Config] Handle $rectorConfig->import() with wildcards * config

### LINKED ISSUE #8151: `RectorConfig::import` no longer supports globs after `0.18`
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.18.0             |

A breaking change was introduced in `0.18.0` where `RectorConfig::import` no longer seems to support globs. Whenever a glob is used, it fails with the following error message:

```
[ERROR] The file "<path>/*" does not exist.
```

This issue affects https://github.com/FriendsOfShopware/shopware-rector/issues/22

## Minimal PHP Code Causing Issue

Since this involves multiple config files, I couldn't reproduce it on the demo page, but I've prepared a very simple MRE on this repository: https://github.com/rmobis/shopware-rector-mre/tree/rector/rector

Cloning the repo, installing composer dependencies and running `vendor/bin/rector process src --dry-run` reproduces the issue.

## Expected Behaviour

When given a glob, rector should load all files matching that glob. For the MRE presented, that means it should report both `RecastingRemovalRector` and `RemoveAlwaysTrueIfConditionRector` issues on the given code.

This was the behaviour on `0.17`. If that is not desired, a breaking change notice should be added to the `0.18` release notes.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8151

Hi @rmobis, this should handle `$rectorConfig->import()` with wildcards * config. Could you verify this? Thank you.

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba I added e2e test for wilcards path import config to prove it :) https://github.com/rectorphp/rector-src/pull/4832/commits/43d4bbfa6d683696348d21f0773a87045dd5613c

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by rmobis** [comment]
@samsonasik I'm afraid the current change doesn't seem to fix the issue completely, only the somewhat incomplete MRE I had presented. I've updated the repository and the issue with appropriate description and instructions to reproduce, but in short the proposed code only imports _one_ file matching the wildcard and not all of them, which is the expected behavior.

PS: I could not find a good way to directly import the branch as a composer dependency (I'm guessing because there's a build step?), so I applied the very small changes manually.

File: packages/Config/RectorConfig.php [hunk 1]
```diff
@@ -200,6 +201,11 @@ static function (
 
     public function import(string $filePath): void
     {
+        $paths = [$filePath];
+
+        $filesystemTweaker  = new FilesystemTweaker();
+        [$filePath] = $filesystemTweaker->resolveWithFnmatch($paths);
```
Comment (staabm) [inline packages/Config/RectorConfig.php]: ~~could we do this only when the filePath contains a `*`?~~

ahh the FilesystemTweaker seems already be smart enough

Comment (staabm) [inline packages/Config/RectorConfig.php]: wouldn't this line - as is - resolve the `*` but picks only the first paths and leaving all others behind?

Comment (samsonasik) [reply packages/Config/RectorConfig.php]: it currently only get single file, I will update to import multiple files in dir.

Comment (samsonasik) [reply packages/Config/RectorConfig.php]: @staabm I updated to imports multiple configs https://github.com/rectorphp/rector-src/pull/4832/commits/eae8bf6743fa7443e6083570877f74263af29cbf

**Comment by samsonasik** [comment]
@rmobis @staabm I updated to allow imports multiple configs resolved by `glob()` https://github.com/rectorphp/rector-src/pull/4832/commits/eae8bf6743fa7443e6083570877f74263af29cbf

**Comment by samsonasik** [comment]
@TomasVotruba @rmobis @staabm I updated e2e test for multi configs included test for the prove :) https://github.com/rectorphp/rector-src/pull/4832/commits/29cae3d1de0b09319336bd3d34758dde285877ea

**Comment by rmobis** [comment]
Seems to be working now. Thanks for the quick fix.

**Comment by TomasVotruba** [comment]
You change my mind with easyfix :) ... Let's ship it then :+1:

**Comment by TomasVotruba** [comment]
In hindsight, we're moving away from magic `*` paths and here is not much gain in making API so open.

Let's keep Rector simple and explicit, like other PHP CLI tools are :+1: 

Reverting here: https://github.com/rectorphp/rector-src/pull/5010

**Comment by rmobis** [comment]
@TomasVotruba Fair enough. However, may I suggest we add a mention to this breaking change in the release notes? Either 0.18 or 0.18.3. Just so that people can easily find out why a previously working behavior now breaks.

**Comment by TomasVotruba** [comment]
I'll update the 0.18 notes, thanks for feedback

**Pull request merged**